### DISCUSSION TITLE
为chirpy添加docker部署支持

### DISCUSSION THREAD

**Discussion by ghost** [cotes2020/jekyll-theme-chirpy#2041] [Ideas]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### Is your feature request related to a problem? Please describe

有部分人与我一样只是为了有一个自己记录笔记的站点，而不想关注具体的构建，虽然GitHubpages很好的满足了这个需求，但是在中国大陆 的访问速度实在太慢，使用vercel等等也无济于事。

### Describe the solution you'd like

自己有服务器的话可以很方便的从docker拉取构建博客所需要的环境，并且一键部署。这里我写了一篇教程，以及我构建这个镜像的过程，以及发布博客的一些小tips。
[部署教程](https://blog.masterke.cn/posts/Automatically_detect_and_rebuild_the_blog_after_pushing_git/)
[docker容器地址](https://hub.docker.com/repository/docker/masterke2003/jekyll-theme-chirpy/general)
已经实现的功能：
1. fork后把代码git拉到指定目录，运行容器即可
2. [有新的代码push自动重新构建](https://blog.masterke.cn/posts/Automatically_detect_and_rebuild_the_blog_after_pushing_git/)
3. [交互式输入自动生成post模板](https://blog.masterke.cn/posts/How_to_gracefully_add_a_new_blog_post/)，自动翻译生成标题名，自动写入当前时间，全身心投入内容
4. [其他教程](https://blog.masterke.cn/tags/jekyll/)

### Describe alternatives you've considered

无

### Additional context

无

**Comment by ghost** [comment]
需要我把这个构建方法写到get start文档里吗？我可以为你提交Pull Request

**Comment by kungfux** [comment]
We have a place where we collect ideas to share with the community that work well for your particular case. Let me move it over there for you.

**Reply by ghost** [reply]
thank you

**Comment by cotes2020** [comment]
Chirpy already supports **Devcontainer**, and you can run the project in a Docker container via CLI. 

Additionally, I don't think the temporary Dockerfile you wrote will be more reliable than the image maintained by Microsoft.

**Reply by ghost** [reply]
Thank you, I am a beginner and have never used Devcontainer before, haha, I have learned something new

**Comment by bestZwei** [comment]
使用 cloudflare pages 部署很快。你看我的 https://zwei.de.eu.org
